core/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -409,6 +422,9 @@ private class InProcessServerStream implements ServerStream {
       final StatsTraceContext statsTraceCtx;
       @GuardedBy("this")
```

Comment:
I had been referring to just the clientNotifyStatus/clientNotifyTrailers before. `setListener()` is set early before anything else (e.g., we don't check for null already) and doesn't change, so this guard is safe to remove. We had it GuardedBy because callbacks needed to run under the lock. Remove this GuardedBy and replace it with a comment about callbacks needing to run in the syncContext (the comment on syncContext can move here). At that point, you don't need to save the value in local variables.

Ditto for serverStreamListener.